[18.0][MIG] agreement_sale: Migration to 18.0

LucasTran380381 opened the pull request.

ygol committed:
[MIG] agreement: Migration to 12.0

bealdav committed:
FIX security issue + details

bealdav committed:
PORT agreement_sale

bealdav committed:
FIX agreement_sale: oca remarks

max3903 committed:
Update ir.model.access.csv

max3903 committed:
Update __manifest__.py

max3903 committed:
[FIX] Missing newline

liweijie0812 committed:
[IMP] move some notions from agreement_legal to agreement

we move the is_template field definition and the agreement.type model from
the agreement_legal module to the agreement module.

The fields are not displayed by default, unless the feature is enabled through a
technical feature group, this is configurable in the agreement_sale module (because agreement
in itself has no UI, and agreement_legal enables the feature by default)

liweijie0812 committed:
Added translation using Weblate (Chinese (Simplified))

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (2 of 2 strings)

Translation: contract-12.0/contract-12.0-agreement_sale
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-agreement_sale/zh_CN/

gurneyalex committed:
[IMP] move some notions from agreement_legal to agreement

we move the is_template field definition and the agreement.type model from
the agreement_legal module to the agreement module.

The fields are not displayed by default, unless the feature is enabled through a
technical feature group, this is configurable in the agreement_sale module (because agreement
in itself has no UI, and agreement_legal enables the feature by default)

Commit:
[MIG][13.0] agreemnt_sale + black

Commit:
[IMP] agreement,agreement_legal: support activities

- The `agreement` model now features mail and activities from its base module.
- The `agreement_legal` module doesn't need to add mail support now.
- Add activities to form views.

@Tecnativa TT22477

sergio-teruel committed:
[IMP] agreement: Minor improvements

sergio-teruel committed:
[ADD] agreement_rebate: New module to manage agreement rebate sales

alexis-via committed:
Re-introduce module agreement_account

Update agreement and agreement_sale to restore the behavior of v10

sergio-teruel committed:
[IMP] agreement: black, isort, prettier

sergio-teruel committed:
[MIG] agreement: Migration to 13.0

newtratip committed:
[IMP] agreement: black, isort, prettier

newtratip committed:
[MIG] agreement: Migration to 14.0

Commit:
[MIG] agreement_sale: Migration to 14.0

Commit:
[IMP] agreement_sale: black, isort, prettier

ibuioli committed:
Added translation using Weblate (Spanish (Argentina))

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (13 of 13 strings)

Translation: contract-14.0/contract-14.0-agreement_sale
Translate-URL: https://translation.odoo-community.org/projects/contract-14-0/contract-14-0-agreement_sale/es_AR/

carlosdauden committed:
[IMP] agreement_rebate: Add active fields in settlements
TT34871

sergio-teruel committed:
[IMP] Add partner_id as deafult _settlement_line_break_fields

sergio-teruel committed:
[IMP] Add partner_id to settlement lines tree view

sergio-teruel committed:
[IMP] agreement_rebate: Set partner_id to settlement lines

carlosdauden committed:
[IMP] agreement_rebate: Improvements

olgamarcocb committed:
[MIG] agreement: Migration to 15.0

etobella committed:
[IMP] agreement: black, isort, prettier

sergio-teruel committed:
[IMP] agreement_rebate: black, isort

sergio-teruel committed:
[MIG] agreement_rebate: Migration to v15.0

Commit:
[IMP] agreement: Change menus structure to make it easy to adapt

Commit:
[IMP] agreement: Change menus structure to make it easy to adapt

FrancoMaxime committed:
[MIG] agreement: Migration to 16.0

FrancoMaxime committed:
[IMP] agreement - Change domain field into computed store field instead of onchange field

ao-landoo committed:
[MIG] agreement_rebate: Migration to 16.0

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (116 of 116 strings)

Translation: agreement-16.0/agreement-16.0-agreement_rebate
Translate-URL: https://translation.odoo-community.org/projects/agreement-16-0/agreement-16-0-agreement_rebate/es/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: contract-14.0/contract-14.0-agreement_sale
Translate-URL: https://translation.odoo-community.org/projects/contract-14-0/contract-14-0-agreement_sale/

Ivorra78 committed:
Added translation using Weblate (Spanish)

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (13 of 13 strings)

Translation: contract-14.0/contract-14.0-agreement_sale
Translate-URL: https://translation.odoo-community.org/projects/contract-14-0/contract-14-0-agreement_sale/es/

pilarvargas-tecnativa committed:
[FIX] aggrement_rebate: The type of movement is defined in the field "move_type"

TT47494

mymage committed:
Added translation using Weblate (Italian)

ramiadavid committed:
[IMP] agreement: pre-commit auto fixes

ramiadavid committed:
[MIG] agreement: Migration to 17.0

imlopes committed:
[IMP] agreement_sale: pre-commit auto fixes

imlopes committed:
[17.0][MIG] agreement_sale: Migration to 17.0

max3903 committed:
Update agreement/models/agreement.py

Co-authored-by: Denis Roussel (ACSONE) <rousseldenis@users.noreply.github.com>

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (64 of 64 strings)

Translation: agreement-17.0/agreement-17.0-agreement
Translate-URL: https://translation.odoo-community.org/projects/agreement-17-0/agreement-17-0-agreement/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (66 of 66 strings)

Translation: agreement-17.0/agreement-17.0-agreement
Translate-URL: https://translation.odoo-community.org/projects/agreement-17-0/agreement-17-0-agreement/it/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: agreement-17.0/agreement-17.0-agreement_sale
Translate-URL: https://translation.odoo-community.org/projects/agreement-17-0/agreement-17-0-agreement_sale/

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (12 of 12 strings)

Translation: agreement-17.0/agreement-17.0-agreement_sale
Translate-URL: https://translation.odoo-community.org/projects/agreement-17-0/agreement-17-0-agreement_sale/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (66 of 66 strings)

Translation: agreement-17.0/agreement-17.0-agreement
Translate-URL: https://translation.odoo-community.org/projects/agreement-17-0/agreement-17-0-agreement/it/

Commit:
[IMP] agreement_sale: pre-commit execution

Commit:
[MIG] agreement_sale: Migration to 18.0

xaviedoanhduy reviewed (commented):
please push a commit to add `test-requirements.txt` for module `agreement` like this to make the CI green:
https://github.com/xaviedoanhduy/bank-statement-import/pull/4/commits/180fce9e6912ee02308f37a496dd4e392b67c5dd

LucasTran380381 commented:
> please push a commit to add `test-requirements.txt` for module `agreement` like this to make the CI green: [xaviedoanhduy/bank-statement-import@180fce9](https://github.com/xaviedoanhduy/bank-statement-import/commit/180fce9e6912ee02308f37a496dd4e392b67c5dd)

will update test-requirement after PR module `agreement` OCA side

xaviedoanhduy commented:
> > please push a commit to add `test-requirements.txt` for module `agreement` like this to make the CI green: [xaviedoanhduy/bank-statement-import@180fce9](https://github.com/xaviedoanhduy/bank-statement-import/commit/180fce9e6912ee02308f37a496dd4e392b67c5dd)
> 
> will update test-requirement after PR module `agreement` OCA side

with this module you can do that, but with other modules that have test files you have to add `test-requirements.txt` so that the dependencies are complete before running the tests files

LucasTran380381 commented:
> > > please push a commit to add `test-requirements.txt` for module `agreement` like this to make the CI green: [xaviedoanhduy/bank-statement-import@180fce9](https://github.com/xaviedoanhduy/bank-statement-import/commit/180fce9e6912ee02308f37a496dd4e392b67c5dd)
> > 
> > 
> > will update test-requirement after PR module `agreement` OCA side
> 
> with this module you can do that, but with other modules that have test files you have to add `test-requirements.txt` so that the dependencies are complete before running the tests files

So for this internal PR, not need to add test-requirement.txt ( I will include it when push PRs of dependencies in OCA side ). Is it right ?

xaviedoanhduy commented:
hi @LucasTran380381, there may be a little misunderstanding. You must know that if a module depends on another module, the dependent module must be installed before this module. The context is that if this module has tests, for example, we must install the dependencies successfully, then the module will start to be installed next. And the rest of CI will start running the tests command to test this module. Fortunately, this module currently does not have test files, so if the dependencies are installed successfully, CI will be green. In short, if you want the tests in a module to run correctly, you must install that module successfully. Otherwise, if it is not installed, it is considered that we have skipped the tests.

you have to make sure that there are no errors when installing so please add that commit so that CI is green indicating that the installation of this module did not have errors

Commit:
[IMP] agreement: pre-commit execution

Commit:
[MIG] agreement: Migration to 18.0

OCA-git-bot committed:
Merge PR #50 into 18.0

Signed-off-by simahawk

Commit:
[UPD] Update agreement.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: agreement-18.0/agreement-18.0-agreement
Translate-URL: https://translation.odoo-community.org/projects/agreement-18-0/agreement-18-0-agreement/

sergio-teruel committed:
[FIX] agreement_rebate: Settlements are excluded if any line has been invoiced before
TT52964

sergio-teruel committed:
[FIX] agreement_rebate: Settlement lines already invoiced are invoiced twice when process the settlement in tow steps
TT52964

CarlosRoca13 committed:
[IMP] agreement_rebate: pre-commit auto fixes

CarlosRoca13 committed:
[MIG] agreement_rebate: Migration to 18.0

OCA-git-bot committed:
[IMP] update dotfiles

OCA-git-bot committed:
Merge PR #56 into 18.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update agreement.pot

Commit:
[UPD] Update agreement_rebate.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: agreement-18.0/agreement-18.0-agreement
Translate-URL: https://translation.odoo-community.org/projects/agreement-18-0/agreement-18-0-agreement/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: agreement-18.0/agreement-18.0-agreement_rebate
Translate-URL: https://translation.odoo-community.org/projects/agreement-18-0/agreement-18-0-agreement_rebate/

OCA-git-bot committed:
[UPD] addons table in README.md

victoralmau committed:
[IMP] agreement_rebate: Improve test to create TestAgreementRebateBase

victoralmau committed:
[IMP] agreement_rebate: Define the sequence without company

ivantodorovich committed:
[REF] agreement_rebate: invoice creation revamp

1. Use `default_*` context keys instead of custom ones.
   This actually solves an issue with the CI.

2. Drop the virtual records to prepare invoice values. That was needed to play
   onchanges but now all interesting fields are computed.

3. Refactor the invoice creation method to process groups in a cleaner way.

OCA-git-bot committed:
Merge PR #84 into 18.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update agreement.pot

Commit:
[UPD] Update agreement_rebate.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: agreement-18.0/agreement-18.0-agreement
Translate-URL: https://translation.odoo-community.org/projects/agreement-18-0/agreement-18-0-agreement/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: agreement-18.0/agreement-18.0-agreement_rebate
Translate-URL: https://translation.odoo-community.org/projects/agreement-18-0/agreement-18-0-agreement_rebate/

OCA-git-bot committed:
Merge PR #79 into 18.0

Signed-off-by pedrobaeza

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (114 of 114 strings)

Translation: agreement-18.0/agreement-18.0-agreement_rebate
Translate-URL: https://translation.odoo-community.org/projects/agreement-18-0/agreement-18-0-agreement_rebate/it/
